step.yml

```diff
@@ -30,58 +34,84 @@ description: |-
-      description: |
-        Path to the given target's Info.plist file. You need to use this Step for each archivable target of your project.
+      title: Project path
+      summary: Xcode Project (`.xcodeproj`) or Workspace (`.xcworkspace`) path
+      description: |-
+        Xcode Project (`.xcodeproj`) or Workspace (`.xcworkspace`) path.
+      is_required: true
 
-        **NOTE:**<br/>
-        If your IPA contains multiple build targets, they would need to have the same version number as your app's main target has.<br/>
-        You need to add this Step to your Workflow for each build target: if you have, say, three targets, you need to have three instances of this Step in your Workflow.
-        If there are targets with different version numbers the app cannot be submitted for App Review or Beta App Review.
+  - scheme: $BITRISE_SCHEME
+    opts:
+      title: Scheme
+      summary: Xcode Scheme name.
+      description: |-
+        Xcode Scheme name.
       is_required: true
+
+  - target:
+    opts:
+      title: Target
+      summary: Xcode Target name.
+      description: |-
+        Xcode Target name.
+        
+        It is optional and if specified then the step will find the given target and update the version numbers for it. 
+        
+        If it is left empty then the step will use the scheme's default target to update the version numbers.
+
+  - configuration:
+    opts:
+      title: Configuration
+      summary: Xcode Configuration name.
+      description: |-
+        Xcode Configuration name.
+        
+        It is optional and if specified then the step will only update the configuration with the given name. 
+        
+        If it is left empty then the step will update all of the target's configurations with the build and version number.
+
   - build_version: "$BITRISE_BUILD_NUMBER"
     opts:
-      title: "Build Number"
-      description: |
-        Set the CFBundleVersion to this value.
-        You can find this in Xcode:
-        - Select your project in the **Project navigator**
-        - Go to the **General** tab and then the **Identity** section
-        - **Build field**
+      title: Build Number
+      description: |-
+        This will be either the CFBundleVersion in the Info.plist file or the CURRENT_PROJECT_VERSION in the project file.
       is_required: true
+
   - build_version_offset:
     opts:
-      title: "Build Number Offset"
-      description: |
+      title: Build Number Offset
+      description: |-
         This offset will be added to `build_version` input's value.
+
   - build_short_version_string:
     opts:
-      title: "Version Number"
-      description: |
-        Set the CFBundleShortVersionString to this value.
-        You can find this in Xcode:
-        - Select your project in the **Project navigator**
-        - Go to the **General** tab and then the **Identity** section
-        - **Version field**
+      title: Version Number
```

Comment:
This is an optional field, what happens when it's not defined?

Reply:
Thanks for reminding me that I did not handle the empty case. So if it is empty then the existing value will not be modified. I also added this to the step.yml.